apps/application-system/api/src/app/modules/application/utils/tokenUtils.ts

```diff
@@ -0,0 +1,16 @@
+import decode from 'jwt-decode'
+import { ExecutionContext } from '@nestjs/common'
+
+export function getNationalIdFromToken(
+  ctx: ExecutionContext,
+): string | undefined {
+  const request = ctx.switchToHttp().getRequest()
+  try {
+    const decodedToken = decode(
+      request.headers.authorization?.replace('Bearer ', ''),
+    ) as { nationalId?: string }
+    return decodedToken?.nationalId
+  } catch (e) {
+    return undefined
```

Comment:
`@NationalId() nationalId: string | undefined`

If decodedToken did not contain a nationalId the user is not authorized. Should that not throw an error?

And should the decorator not always be written like this:
`@NationalId() nationalId: string`

So the developer is certain that if the handler in the controller runs, then the user has a valid token.

Reply:
true!